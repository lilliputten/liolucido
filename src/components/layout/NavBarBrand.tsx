'use client';

import Image from 'next/image';
import { usePathname } from 'next/navigation';
import { useLocale } from 'next-intl';
import { useTheme } from 'next-themes';

import { siteTitle } from '@/config/env';
import { infoRoute, welcomeRoute } from '@/config/routesConfig';
import { getAllRouteSynonyms } from '@/lib/routes';
import { TPropsWithChildrenAndClassName } from '@/lib/types';
import { cn } from '@/lib/utils';
import logoSvg from '@/assets/logo/logo-full-line-w.svg';
import logoWhiteSvg from '@/assets/logo/logo-full-line-w.svg';
import { isDev } from '@/constants';
import { Link } from '@/i18n/routing';
import { TLocale } from '@/i18n/types';

interface NavBarBrandProps {
  isUser?: boolean;
}

function BrandWrapper(props: TPropsWithChildrenAndClassName & NavBarBrandProps) {
  const { isUser, children, className: parentClassName } = props;
  const locale = useLocale() as TLocale;
  const pathname = decodeURI(usePathname() || '');
  const rootRoute = isUser ? infoRoute : welcomeRoute;
  const rootRoutesList = getAllRouteSynonyms(rootRoute, locale);
  const isRoot = !pathname || rootRoutesList.includes(pathname);
  const className = cn(
    isDev && '__BrandWrapper', // DEBUG
    parentClassName,
    'flex',
    'items-center',
    'space-x-1.5',
    'gap-2',
    'transition-all',
    'mr-10',
    'select-none',
    !isRoot && 'hover:opacity-80',
  );
  if (isRoot) {
    return <div className={className}>{children}</div>;
  }
  return (
    <Link href={rootRoute} className={className}>
      {children}
    </Link>
  );
}

export function NavBarBrand(props: NavBarBrandProps) {
  const { resolvedTheme } = useTheme();
  const isDark = resolvedTheme === 'dark';
  return (
    <BrandWrapper {...props} className="h-12">
      <Image
        src={isDark ? logoWhiteSvg : logoSvg}
        className="h-14 w-auto"
        priority={false}
        alt={siteTitle}
      />
      {/*
      <h1
        role="heading"
        data-testid="NavBarBrandTitle"
        className={cn(
          'font-urban',
          'text-xl',
          'text-theme-foreground',
          'font-bold',
          'whitespace-nowrap',
          'overflow-hidden',
          'text-ellipsis',
        )}
      >
        {siteTitle}
      </h1>
      */}
    </BrandWrapper>
  );
}
