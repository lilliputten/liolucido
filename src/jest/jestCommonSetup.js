/* eslint-env jest */

/** Test db. Ensure if it has been created */
const DATABASE_URL = 'file:.data/test.db';

// Set it for prisma also
process.env.DATABASE_URL = DATABASE_URL;
