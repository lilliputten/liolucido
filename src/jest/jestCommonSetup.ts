/* eslint-env jest */

import { ExtendedUser } from '@/@types/next-auth';

/* // NOTE: Using the local db for tests. Make sure all the data is cleaning.
 * // Test db. Ensure if it has been created
 * const DATABASE_URL = 'file:.data/test.db';
 * // Set it for prisma also
 * process.env.DATABASE_URL = DATABASE_URL;
 */

// Mocks...

jest.mock('@/jest/test/bare', () => ({
  getBare: jest.fn(() => 'initial mocked bare'),
}));

jest.mock('@/lib/session', () => ({
  getCurrentUser: jest.fn() as jest.MockedFunction<() => Promise<ExtendedUser | undefined>>,
}));
