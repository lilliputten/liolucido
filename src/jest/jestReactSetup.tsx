import React from 'react';

import { pathnames } from '@/constants/routes';

jest.mock('next-intl', () => ({
  useLocale() {
    return 'en';
  },
}));

interface TLinkProps {
  children: React.ReactNode;
  href: string;
  className?: string;
  [key: string]: unknown;
}

jest.mock('@/i18n/routing', () => ({
  Link: (props: TLinkProps) => {
    const { children, href, className, ...restProps } = props;
    return React.createElement('a', { href, className, ...restProps }, children);
  },
  pathnames,
}));
