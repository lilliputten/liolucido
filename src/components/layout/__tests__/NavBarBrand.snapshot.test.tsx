/** @jest-environment jsdom
 */

import '@/jest/jestDomSetup';
import '@/jest/jestReactSetup';

import { render } from '@testing-library/react';

import { NavBarBrand } from '../NavBarBrand';

it('NavBarBrand snapshot', () => {
  const { container } = render(<NavBarBrand isUser={false} isUserRequired={false} />);
  expect(container).toMatchSnapshot();
});
