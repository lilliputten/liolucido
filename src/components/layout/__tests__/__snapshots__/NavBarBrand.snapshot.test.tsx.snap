// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarBrand snapshot 1`] = `
<div>
  <div
    class="h-12 flex items-center space-x-1.5 gap-2 transition-all mr-10 select-none"
  >
    <img
      alt="Logo"
      class="size-16"
      data-nimg="1"
      decoding="async"
      height="36"
      loading="lazy"
      src="/static/logo/Logo.svg"
      style="color: transparent;"
      width="36"
    />
    <h1
      class="font-urban text-xl text-theme-foreground font-bold whitespace-nowrap overflow-hidden text-ellipsis"
      data-testid="NavBarBrandTitle"
      role="heading"
    >
      TrainWizzz!
    </h1>
  </div>
</div>
`;
